packages/bindgen/src/templates/node-wrapper.ts

```diff
@@ -0,0 +1,221 @@
+    // TODO consider importing instead.
+    // Might be slightly faster to make dedicated wrapper for 1 and 2 argument forms, but unlikely to be worth it.
+    function _promisify(func) {
+      return new Promise((resolve, reject) => {
+        func((...cbargs) => {
+          if (cbargs.length < 1 || cbargs.length > 2) throw Error("invalid cbargs length " + cbargs.length);
+          let error = cbargs[cbargs.length - 1];
+          if (error) {
+            reject(error);
+          } else if (cbargs.length == 2) {
+            const result = cbargs[0];
+            if (result === null || result === undefined) {
+              throw new Error("Unexpected null or undefined successful result");
+            }
+            resolve(result);
+          } else {
+            resolve();
+          }
+        });
+      });
+    }
+  `);
+
+  const injectables = [
+    "Long",
+    "ArrayBuffer",
+    "Float",
+    "ObjectId",
+    "UUID",
+    "Decimal128",
+    "EJSON_parse: EJSON.parse",
+    "EJSON_stringify: EJSON.stringify",
+  ];
+
+  for (const cls of spec.classes) {
+    injectables.push(cls.jsName);
+
+    let body = "";
+    let body_browser = "";
+
+    // It will always be accessed via this name rather than a static to enable optimizations
+    // that depend on the symbol not changing.
+    const symb = `_${cls.rootBase().jsName}_Symbol`;
+
+    if (!cls.base) {
+      // Only root classes get symbols and constructors
+      all(`const ${symb} = Symbol("Realm.${cls.jsName}.external_pointer");`);
+      browserLines.push(
+        `const _${cls.jsName}_registery = new FinalizationRegistry(nativeModule.${cls.jsName}_deleter);`,
+      );
+
+      body += `constructor(ptr) { this[${symb}] = ptr; };`;
+      body_browser += `constructor(ptr) { this[${symb}] = ptr; _${cls.jsName}_registery.register(this, ptr);};`;
+    }
+
+    // This will override the extractor from the base class to do a more specific type check.
+    const extract_definition = `
+      static _extract(self) {
+        if (!(self instanceof ${cls.jsName}))
+          throw new TypeError("Expected a ${cls.jsName}");
+        const out = self[${symb}];
+        if (!out)
+          throw new TypeError("received an improperly constructed ${cls.jsName}");
+        return out;
+      };
+    `;
+    body += extract_definition;
+    body_browser += extract_definition;
+
+    for (const method of cls.methods) {
+      // Eagerly bind the name once from the native module.
+      const native = `_native_${method.id}`;
+      all(`const ${native} = nativeModule.${method.id};`);
+      // TODO consider pre-extracting class-typed arguments while still in JIT VM.
+      const asyncSig = method.sig.asyncTransform();
+      const params = (asyncSig ?? method.sig).args.map((a) => a.name);
+      const args = [
+        method.isStatic ? [] : `this[${symb}]`, //
+        ...params,
+        asyncSig ? "_cb" : [],
+      ].flat();
+      let call = `${native}([${args}])`;
```

Comment:
I didn't notice this change at first. Please change this back to just `(${args})`. We shouldn't be making a new JS Array every time we call a native function!

Reply:
It's easier to access the args consistently this way in the generated C++ (we don't have a JS engine abstraction like in node and JSI that facilitate accessing the args). Also passing var args is not supported AFAIK from the `Embind` perspective.

Reply:
You don't need to use varargs in C++. When generating the C++, we know exactly how many arguments are going to be passed, so we can just generate a function taking that many args. You can choose between naming the args `arg1, arg2,....` so you just need to change `arg[${i}]` to `arg${i}`, or you can just use the arg names from the spec (`a.name`, with `'_self'` for the `this` parameter in non-static methods). Slight preference for using the meaningful names, but not a big deal either way.

Reply:
You won't be able to check the number of passed args for error handling, but that is probably fine. I think we can get rid of that everywhere since it is probably unnecessary now that we have the generated JS code to ensure that we always pass the correct number of arguments (although some may be `undefined`)